lib/sablon/environment.rb

```diff
@@ -0,0 +1,47 @@
+module Sablon
+  # Combines the user supplied context and template into a single object
+  # to manage data during template processing.
+  class Environment
+    attr_reader :template
+    attr_reader :context
+
+    # returns a new environment with merged contexts
+    def alter_context(context = {})
+      new_context = @context.merge(context)
+      Environment.new(@template, new_context)
+    end
+
+    private
+
+    def initialize(template, context = {})
+      @template = template
+      @context = transform_hash(context)
+    end
+
+    def transform_hash(hash)
```

Comment:
Could we keep those functions in the `Context` module to make it explicit that they are about the context not the environment?